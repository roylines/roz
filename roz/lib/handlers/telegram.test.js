const {handle} = require('./telegram');
const {lambda} = require('../aws');
const {warn} = require('lambda-log');

jest.mock('../aws');
jest.mock('lambda-log');

describe('telegram handler', () => {
  const context = {
    invokedFunctionArn: 'ARN',
  };
  process.env.TELEGRAM_USER = 42;
  it('should warn and return 200 for unknown user', async () => {
    warn.mockReset();
    const body = {message: {from: {id: 84}}};
    const response = await handle({body: JSON.stringify(body)}, context);

    expect(lambda.invoke).not.toHaveBeenCalled();
    expect(response).toMatchSnapshot({statusCode: 200});
    expect(warn.mock.calls[0][0]).toEqual('unauthorised user');
  });

  it('should invoke for known user', async () => {
    const body = {message: {text: 'hello', from: {id: 42}}};
    const response = await handle({body: JSON.stringify(body)}, context);

    expect(response).toMatchSnapshot({statusCode: 200});
    expect(lambda.invoke.mock.calls).toMatchSnapshot();
  });
});
