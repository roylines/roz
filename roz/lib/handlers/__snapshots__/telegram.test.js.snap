// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`telegram handler should invoke for known user 1`] = `
Object {
  "statusCode": 200,
}
`;

exports[`telegram handler should invoke for known user 2`] = `
Array [
  Array [
    Object {
      "arn": "ARN",
      "message": Object {
        "from": Object {
          "id": 42,
        },
        "text": "hello",
      },
    },
  ],
]
`;

exports[`telegram handler should warn and return 200 for unknown user 1`] = `
Object {
  "statusCode": 200,
}
`;
